postprocessors/sandbox/widebeam_downsample2normalscan.py

```diff
@@ -0,0 +1,217 @@
+                raise conversion_exceptions.ImproperFileStructureError(
+                    f'Input file structure "{self.infile_structure}" is not compatible with input file type '
+                    f'"{self.infile_type}": Valid structures for {self.infile_type} are '
+                    f'{rs.FILE_STRUCTURE_MAPPING[self.infile_type]}'
+                )
+            if self.outfile_structure not in rs.FILE_STRUCTURE_MAPPING[self.outfile_type]:
+                raise conversion_exceptions.ImproperFileStructureError(
+                    f'Output file structure "{self.outfile_structure}" is not compatible with output file type '
+                    f'"{self.outfile_type}": Valid structures for {self.outfile_type} are '
+                    f'{rs.FILE_STRUCTURE_MAPPING[self.outfile_type]}'
+                )
+            if self.infile_structure not in ['array', 'site', 'dmap']: #added to accept dmap input
+                raise conversion_exceptions.ConversionUpstreamError(
+                    f'Input file structure "{self.infile_structure}" cannot be reprocessed into any other format.'
+                )
+        check_args(self)
+        self.averaging_method = None
+        self._temp_files = []
+        self.process_file()
+
+    @staticmethod
+    def process_record(record: OrderedDict, **kwargs) -> OrderedDict:
+        """
+        Takes a record from a rawacf file process into a rawacf record.
+        This method also keeps a single designated beam,
+
+        Parameters
+        ----------
+        record: OrderedDict
+            hdf5 record containing rawacf data and metadata
+        beam_index: Dict
+            A dictionary mapping timestamp to beam index
+
+        Returns
+        -------
+        record: OrderedDict
+            hdf5 record, downsampled to one beam
+        """
+        beam2keep = 0
+        first_min = dt.datetime.fromtimestamp(record['sqn_timestamps'][0]).replace(second =0, microsecond = 0)
+        timestamp = dt.datetime.fromtimestamp(record['sqn_timestamps'][0])
+        diff = abs(first_min - timestamp).total_seconds()/record['int_time']
+        beam2keep = int(round(diff))
+        if beam2keep >= 16:
+            beam2keep = 0
+
+        #Seperate the beam to keep
+        record['beam_nums'] = np.array([np.uint32(beam2keep)])
+        record['beam_azms'] = np.array([record['beam_azms'][beam2keep]])
+        try:
+            record['data_dimensions'][0] = 1
+        except:
+            record['correlation_dimensions'][0] = 1
+        record['main_acfs'] = record['main_acfs'][beam2keep, :, :].reshape(1, record['data_dimensions'][1], record['data_dimensions'][2])
+        record['intf_acfs'] = record['intf_acfs'][beam2keep, :, :].reshape(1, record['data_dimensions'][1], record['data_dimensions'][2])
+        record['xcfs'] = record['xcfs'][beam2keep, :, :].reshape(1, record['data_dimensions'][1], record['data_dimensions'][2])
```

Comment:
```suggestion
        if "data_dimensions" in record:
            record['data_dimensions'][0] = 1
        else:
            record['correlation_dimensions'][0] = 1
        record['main_acfs'] = record['main_acfs'][beam2keep, :, :]
        record['intf_acfs'] = record['intf_acfs'][beam2keep, :, :]
        record['xcfs'] = record['xcfs'][beam2keep, :, :]
```
The `reshape()` calls weren't doing anything, as far as I could tell.